mlserver/batching/adaptive.py

```diff
@@ -52,6 +57,10 @@ async def _queue_request(
     ) -> Tuple[str, Awaitable[InferenceResponse]]:
         internal_id = generate_uuid()
 
+        with self.batch_queue_request_count.time():
+            batch_queue_size = self._requests.qsize()
+            self.batch_queue_request_count.observe(batch_queue_size)
```

Comment:
We used Histogram because we needed more sampling items. Otherwise with Gauge we didn't have enough sampling points between two values. The number of elements of the queue depends on time on such a way.